clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -106,12 +106,12 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals,  Map<String, ?>
 
         this.originals = resolveConfigVariables(configProviderProps, (Map<String, Object>) originals);
         this.values = definition.parse(this.originals);
+        this.used = Collections.synchronizedSet(new HashSet<>());
```

Comment:
Just curious is this line necessary for the fix?

Reply:
It's not new-added but moved before the call to `postProcessParsedConfig`, otherwise an NPE exception will be thrown.